pixeltable/exprs/type_cast.py

```diff
@@ -0,0 +1,68 @@
+import json
+from typing import Optional, Dict, List, Tuple, Any
+
+import sqlalchemy as sql
+
+from pixeltable.exprs import DataRow, Expr, RowBuilder
+from pixeltable.type_system import ColumnType
+
+
+class TypeCast(Expr):
+
+    def __init__(self, underlying: Expr, new_type: ColumnType):
+        super().__init__(new_type)
+        if self.col_type.is_string_type() \
+                or self.col_type.is_json_type() and underlying.col_type.is_string_type():
```

Comment:
This check doesn't really work, because a json-typed column can give you a string value.

See docs/tutorials/openai_demo.ipynb (look at column eval_output at the very bottom; it has type json but contains strings).

We want to extend the tutorial to cast t.eval.choices[0].message.content to an actual dict (it already has column type json, but a single string is actually a valid json value; an explicit cast should try to convert the string to a dict).

Reply:
I'm slightly confused by this. What type of cast are you talking about?

If I understand this comment right, you're concerned about the case where the column is a JsonType, and we want to cast it to a JsonType?

I'd argue that a cast from type `T` to `T` should always be a no-op. If it's a JsonType column and it holds a string because a string is valid JSON, then it should still hold a string after the cast.

I'd like to better understand the use case you're concerned about.

Reply:
Note: I realize that the first cut of this didn't allow converting `T` to `T` at all; I've now fixed this (I think)